### ISSUE TITLE
Changes not detected for version bump

### ISSUE THREAD

**Issue by daveenguyen** [relekang/python-semantic-release#135]
running `semantic-release version` in this repo's state https://github.com/daveenguyen/need-psr-help

I am getting

```
Creating new version..
Current version: 1.0.0
No release will be made.
```

I'm expecting it to bump to 2.0.0 because there's a breaking feature commit
https://github.com/daveenguyen/need-psr-help/commit/f0f21c4e2f7ec36af9c9467147b3641ddb65b394

I'm using commitizen to write commits.

**Comment by relekang** [comment]
Can you please run with environment variable `DEBUG=semantic_release.*` and post the logs?

**Comment by daveenguyen** [comment]
```
  semantic_release:cli 'main args:' {'force_level': None, 'post': False, 'retry': False, 'noop': False} +0.0us
  semantic_release:cli 'main env:' 'GH_TOKEN="None",PYPI_USERNAME="None",PYPI_PASSWORD="None",' +760.75us
  semantic_release:cli 'main config:' {'check_build_status': 'false', 'commit_message': 'Automatically generated by python-semantic-release', 'commit_parser': 'semantic_release.history.angular_parser', 'patch_without_tag': 'false', 'upload_to_pypi': 'false', 'version_source': 'commit'} +768.0us
Creating new version..
  semantic_release:history 'get_current_version_by_config_file' +0.0us
  semantic_release:history '__init__.py' '__version__' +604.75us
  semantic_release:history <_sre.SRE_Match object; span=(0, 19), match="__version__='1.0.0'"> +5.4ms
Current version: 1.0.0
  semantic_release:history:logs 'evaluate_version_bump("1.0.0", "None")' +0.0us
  semantic_release:history:logs 'found 1.0.0 in "feat: default to path-in-name to be compatible with 0.x\n\nBREAKING CHANGE: Switch back to default behavior of including path in name. Breaks 1.0.0 but is\ncompatible with 0.x\n. breaking loop' +46.3ms
  semantic_release:history 'get_new_version("1.0.0", "None")' +52.9ms
No release will be made.
```

**Comment by relekang** [comment]
The result after the fix on your repo:
```
$ PYTHONPATH=../semantic-release/ python -m semantic_release.cli version  --noop
Creating new version..
Current version: 1.0.0
No operation mode. Should have bumped from 1.0.0 to 2.0.0.
```

**Comment by relekang** [comment]
Released in v4.1.2
